client/apis/volumesnapshot/v1/types.go

```diff
@@ -0,0 +1,429 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// +kubebuilder:object:generate=true
+package v1
+
+import (
+	core_v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// VolumeSnapshot is a user's request for either creating a point-in-time
+// snapshot of a persistent volume, or binding to a pre-existing snapshot.
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:scope=Namespaced
+// +kubebuilder:subresource:status
```

Comment:
Currently, the storage and served version true is set to the v1beta1. v1 should have `served: true` and `storage: true` and v1beta1 should only have `served: true`. To do that
```suggestion
// +kubebuilder:subresource:status
// +kubebuilder:storageversion
```

Reply:
Thanks @prafull01 for checking this!  We actually have to do this in steps.  In 1.20, we move API to v1 but keep storage version at v1beta1.  In 1.21, we plan to change storage version to v1.  This is because we added the validation webhook in 1.19.  See the following KEP for details.
https://github.com/kubernetes/enhancements/tree/master/keps/sig-storage/1900-volume-snapshot-validation-webhook

Reply:
Thanks @xing-yang for letting me know. I will remove my comments. :+1: